### PR TITLE
give priority to config in b-c

### PR DESCRIPTION
**Pull request by mihaiplesa**
Resolves https://github.com/brave/brave-browser/issues/19499

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "brave-core",
```
Comment (mihaiplesa) [inline package.json:2]: this is used in produced artifact names and it was taken before from b-b and was `brave`, to evaluate impact

File: build/commands/lib/config.js [hunk 2]
```diff
@@ -29,36 +29,17 @@ const run = (cmd, args = []) => {
   return prog
 }
 
-// this is a huge hack because the npm config doesn't get passed through from brave-browser .npmrc/package.json
```
Comment (petemill) [inline build/commands/lib/config.js:32]: why is this removed? Doesn't seem like any of this would work without this function?

Comment (mihaiplesa) [reply build/commands/lib/config.js:32]: oops

File: build/commands/lib/config.js [hunk 3]
```diff
@@ -48,17 +48,17 @@ var packageConfig = function(key, sourceDir = rootDir){
...
 
-const getNPMConfig = (key, fallbackToBraveCoreOnly) => {
+const getNPMConfig = (key) => {
   if (!NpmConfig) {
     const list = run(npmCommand, ['config', 'list', '--json', '--userconfig=' + path.join(rootDir, '.npmrc')])
     NpmConfig = JSON.parse(list.stdout.toString())
   }
 
-  return NpmConfig[key.join('-').replace(/_/g, '-')] || packageConfig(key) || packageConfigBraveCore(key)
+  return NpmConfig[key.join('-').replace(/_/g, '-')] || packageConfigBraveCore(key) || packageConfig(key)
```
Comment (mihaiplesa) [inline build/commands/lib/config.js:61]: swapping this so `version` get priority from b-c

File: script/lib/config.py [hunk 4]
```diff
@@ -81,8 +81,8 @@ def product_name():
 
 
 def get_chrome_version():
-    version = (os.environ.get('npm_config_brave_version') or
-               brave_browser_package()['config']['projects']['chrome']['tag'])
+    version = (os.environ.get('npm_config_chrome_version') or
+               brave_core_package()['config']['projects']['chrome']['tag'])
```
Comment (mihaiplesa) [inline script/lib/config.py:85]: this was missed when Chromium tag got moved from b-b

Comment (petemill) [reply script/lib/config.py:85]: looks like this is used by bump_version script

Comment (mihaiplesa) [reply script/lib/config.py:85]: reverted rename of `npm_config_brave_version` to `npm_config_chrome_version`

**Review by petemill — APPROVED** [review]
so essentially the only code change in this PR is prioritizing brave-core config over brave-browser, which makes sense. Please squash the revert before merge.

**Review by bridiver — APPROVED** [review]
